src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs

```diff
@@ -156,23 +156,19 @@ private static PSCommand BuildPSCommandFromArguments(string command, IReadOnlyLi
                 return new PSCommand().AddCommand(command);
             }
 
-            // We are forced to use a hack here so that we can reuse PowerShell's parameter binding
+            // HACK: We use AddScript instead of AddArgument/AddParameter to reuse Powershell parameter binding logic.
+            // We quote the command parameter so that expressions can still be used in the arguments.
             var sb = new StringBuilder()
                 .Append("& ")
-                .Append(StringEscaping.SingleQuoteAndEscape(command));
+                .Append('"')
+                .Append(command)
+                .Append('"');
 
             foreach (string arg in arguments)
             {
-                sb.Append(' ');
-
-                if (StringEscaping.PowerShellArgumentNeedsEscaping(arg))
-                {
-                    sb.Append(StringEscaping.SingleQuoteAndEscape(arg));
-                }
-                else
-                {
-                    sb.Append(arg);
-                }
+                sb
```

Comment:
Nit: consider,

```csharp
sb.Append(' ').Append(ArgumentEscaping.Escape(arg));
```

Personally I like using `text` as the variable for a string builder to enable this:

```csharp
text.Append(' ')
    .Append(ArgumentEscaping.Escape(arg));
```

Very minor.

Reply:
I was matching style of other things in PSES, we can certainly do this if @andschwa doesn't mind.

Reply:
Totally fine. I'm not super aggressive about style, I think anytime a style question is in play it should be addressed by tooling rather than back and forth on PRs. Do what you think is readable 😄